src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs

```diff
@@ -635,30 +637,33 @@ internal bool IsBestVersion(string name, SemanticVersion version)
             }
         }
 
-        private async Task InstallPackages(List<GraphItem> installItems, string packagesDirectory,
-            Func<Library, string, bool> packageFilter)
+        private async Task InstallPackages(List<GraphItem> installItems, string packagesDirectory)
         {
-            using (var sha512 = SHA512.Create())
+            if (RestoringInParallel())
+            {
+                await ForEachAsync(
+                    installItems,
+                    MaxDegreesOfConcurrency,
+                    item => InstallPackageCore(item, packagesDirectory));
+            }
+            else
             {
                 foreach (var item in installItems)
                 {
-                    var library = item.Match.Library;
-
-                    var memStream = new MemoryStream();
-                    await item.Match.Provider.CopyToAsync(item.Match, memStream);
-                    memStream.Seek(0, SeekOrigin.Begin);
-                    var nupkgSHA = Convert.ToBase64String(sha512.ComputeHash(memStream));
+                    await InstallPackageCore(item, packagesDirectory);
+                }
+            }
+        }
 
-                    bool shouldInstall = packageFilter(library, nupkgSHA);
-                    if (!shouldInstall)
-                    {
-                        continue;
-                    }
+        private async Task InstallPackageCore(GraphItem installItem, string packagesDirectory)
+        {
+            using (var memoryStream = new MemoryStream())
+            {
+                var match = installItem.Match;
+                await match.Provider.CopyToAsync(installItem.Match, memoryStream);
 
-                    Reports.Information.WriteLine("Installing {0} {1}", library.Name.Bold(), library.Version);
-                    memStream.Seek(0, SeekOrigin.Begin);
-                    await NuGetPackageUtils.InstallFromStream(memStream, library, packagesDirectory, sha512);
-                }
+                memoryStream.Position = 0;
```

Comment:
Yes, please revert it. I guess it's better to use the same approach consistently.